docs/my-website/docs/proxy/guardrails/acuvity_ai.md

```diff
@@ -0,0 +1,194 @@
+import Image from '@theme/IdealImage';
+import Tabs from '@theme/Tabs';
+import TabItem from '@theme/TabItem';
+
+# Acuvity Guardrails for LiteLLM
+
+Use **Acuvity** to detect **PII**, **Prompt Injection**, and other security risks in requests and responses.
+
+## 1. Installation
+
+Since Acuvity is an optional dependency, install it using Poetry:
+
+```shell
+poetry install --extras "acuvity"
+```
+
+Alternatively, if you are using `pip`:
+
+```shell
+pip install acuvity
+```
+
+## 2. Setup Guardrails on Acuvity
+
+### On more details on the usage of Acuvity with guardrails, please visit the https://docs.acuvity.ai/
+
+### **Define Guardrails for Different Stages**
+
+Acuvity supports guardrails at different stages of an LLM request:
+
+1. **Pre LLM API Call** - Validate and redact sensitive information before sending the request.
+2. **During LLM API Call** - Monitor for prompt injection or malicious content.
+3. **Post LLM API Call** - Analyze and filter responses for security violations.
+
+
+### **Pre-Call: Detect and Redact PII**
+
+Add the **PII Detection** guardrail to your **Pre LLM API Call** configuration.
+
+
+### **During-Call: Detect Prompt Injection**
+
+Enable **Prompt Injection Detection** for your **During LLM API Call** configuration.
+
+
+### **Post-Call: Monitor Responses for Security Issues**
+
+Configure **Post LLM API Call** guardrails to filter inappropriate or malicious responses.
+
+
+## 3. Define Guardrails in Your LiteLLM `config.yaml`
+
+### On more details on the usage of Acuvity with guardrails, please visit the https://docs.acuvity.ai/
+
+Update your LiteLLM `config.yaml` file to include Acuvity guardrails:
+
+```yaml
+model_list:
+  - model_name: gpt-4
+    litellm_params:
+      model: openai/gpt-4
+      api_key: os.environ/OPENAI_API_KEY
+
+guardrails:
+  - guardrail_name: "acuvity-pre-guard"
+    litellm_params:
+      guardrail: acuvity
+      mode: "pre_call"
+      default_on: true
+      api_key: os.environ/ACUVITY_TOKEN
+      vendor_params:
+        guardrails:
+          - name: pii_detector
+            matches:
+              email_address:
+                redact: true
+  - guardrail_name: "acuvity-during-guard"
+    litellm_params:
+      guardrail: acuvity
+      mode: "during_call"
+      default_on: true
+      api_key: os.environ/ACUVITY_TOKEN
+      vendor_params:
+        guardrails:
+          - name: prompt_injection
+  - guardrail_name: "acuvity-post-guard"
+    litellm_params:
+      guardrail: acuvity
+      mode: "post_call"
+      default_on: true
+      api_key: os.environ/ACUVITY_TOKEN
+      vendor_params:
+        guardrails:
+          - name: prompt_injection
```

Comment:
We probably need a better example here for a post guard. Probably malcontent detection could be useful here? Basically, you want to check if the output is biased, and then trigger a denial. Also, reflect it in the guardrail name. We shouldn't call them according to the underlying method.